File: LeetCode/April Challenge/8th Day/Day_8.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        int n=0;
        ListNode* temp;
        temp=head;
        while(temp != NULL)
        {
            n++;
            temp=temp->next;
        }
        for(int i=1;i<=(n/2);i++)
            head=head->next;
        
        return head;
    }
};

File: LeetCode/April Challenge/8th Day/Day_8.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def middleNode(self, head: ListNode) -> ListNode:
        n = 0
        temp = head
        while temp != None:
            n += 1
            temp = temp.next
        for i in range(n // 2):
            head = head.next
        return head
